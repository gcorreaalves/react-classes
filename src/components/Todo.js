import React from 'react';
import styled from 'styled-components';

import Button from './Button';

const TodoStyled = styled.div`
  font-size: 18px;
  line-height: 1;
  text-decoration: ${({ completed }) => completed ? 'line-through' : ''};
  margin: 0;
  cursor: pointer;
  &:before{
    content: '';
    background: url('${(props) => props.completed
        ? 'data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="-10 -18 100 135"><circle cx="50" cy="50" r="50" fill="none" stroke="#bddad5" stroke-width="3"/><path fill="#5dc2af" d="M72 25L42 71 27 56l-4 4 20 20 34-52z"/></svg>'
        : 'data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" viewBox="-10 -18 100 135"><circle cx="50" cy="50" r="50" fill="none" stroke="#ededed" stroke-width="3"/></svg>'
    }') no-repeat;
    display:inline-block;
    vertical-align:middle;
    height: 40px;
    width: 40px;
    margin-right: 10px;
  }
`;

TodoStyled.displayName = 'TodoStyled';

export default (props) => (
  <div className="clear" style={{margin: "10px 0"}}>
    <TodoStyled
      className="float-left"
      completed={props.completed}
      onClick={props.handleTodoClick}
    >
      {props.children}
    </TodoStyled>
    <Button
      className="float-right"
      danger
      onClick={props.handleTodoDeleteClick}
    >
      Delete
    </Button>
  </div>
);
