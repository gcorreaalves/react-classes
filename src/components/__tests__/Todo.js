import React from 'react';

import Todo from '../Todo';

describe('Todo component tests', () => {
  let todoWrapper;
  
  beforeEach(() => {
    todoWrapper = shallow(<Todo>Test the task</Todo>);
  });

  it('should render a Todo component', () => {
    expect(todoWrapper).toMatchSnapshot();
  });

  it('should show the Todo component text', () => {
    const text = todoWrapper.find('TodoStyled').dive().text();
    expect(text).toBe('Test the task');
  });

  it('should set the Todo component as completed', () => {
    todoWrapper = shallow(<Todo completed>Test the completed task</Todo>);
    const isCompleted = todoWrapper.find('TodoStyled').prop('completed');
    expect(isCompleted).toBeTruthy();
  });

  it('should show a button to remove the Todo component', () => {
    expect(todoWrapper.find('Button').length).toBe(1);
  });

  it('should call handleTodoClick function when clicking on Todo Component', () => {
    const handleTodoClick = jest.fn();
    todoWrapper = shallow(
      <Todo handleTodoClick={handleTodoClick}>
        Test the completed task
      </Todo>
    );
    todoWrapper.find('TodoStyled').simulate('click');
    expect(handleTodoClick).toHaveBeenCalledTimes(1);
  });

  it('should call handleRemoveClick function when clicking on Remove Button', () => {
    const handleTodoDeleteClick = jest.fn();
    todoWrapper = shallow(
      <Todo handleTodoDeleteClick={handleTodoDeleteClick}>
        Test the completed task
      </Todo>
    );
    todoWrapper.find('Button').simulate('click');
    expect(handleTodoDeleteClick).toHaveBeenCalledTimes(1);
  });
});
