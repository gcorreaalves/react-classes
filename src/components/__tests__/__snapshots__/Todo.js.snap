// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component tests should render a Todo component 1`] = `
<div
  className="clear"
  style={
    Object {
      "margin": "10px 0",
    }
  }
>
  <TodoStyled
    className="float-left"
  >
    Test the task
  </TodoStyled>
  <Button
    className="float-right"
    danger={true}
  >
    Delete
  </Button>
</div>
`;
